### PR TITLE
Adjustments

### PR DESCRIPTION
**Pull request by Nightenom**
# Changes proposed in this pull request:
- fixes to latest blockout upgrade

Review please

### CODE REVIEW

File: src/main/java/com/ldtteam/blockout/controls/AbstractTextElement.java [hunk 1]
```diff
@@ -396,20 +403,39 @@ public void setText(final List<IFormattableTextComponent> text)
...
+    }
+
     public void setText(final IFormattableTextComponent text)
     {
-        this.text = Collections.singletonList(text);
-        recalcTextRendering();
+        setText(Collections.singletonList(text));
+    }
+
+    public void clearText()
```
Comment (Raycoms) [inline src/main/java/com/ldtteam/blockout/controls/AbstractTextElement.java:445]: just missing javadoc on those

**Comment by Raycoms** [comment]
:shipit:

**Comment by Nightenom** [comment]
I might add more as I go through mcol code :D

File: src/main/java/com/ldtteam/blockout/Loader.java [hunk 2]
```diff
@@ -275,4 +280,9 @@ public static void addToCache(ResourceLocation loc, PaneParams doc)
 
         parsedCache.put(loc, new Tuple<>(1, doc));
     }
+
+    public static void cleanXMLparseCache()
```
Comment (Raycoms) [inline src/main/java/com/ldtteam/blockout/Loader.java]: jdoc

File: src/main/java/com/ldtteam/blockout/controls/AbstractTextElement.java [hunk 3]
```diff
@@ -268,6 +294,8 @@ else if (textAlignment.isHorizontalCentered())
             lineShift += mc.fontRenderer.FONT_HEIGHT + textLinespace;
         }
 
+        // TODO: forge disable filtering
```
Comment (Raycoms) [inline src/main/java/com/ldtteam/blockout/controls/AbstractTextElement.java:300]: todo for the future?

Comment (Nightenom) [reply src/main/java/com/ldtteam/blockout/controls/AbstractTextElement.java:300]: todo for this <https://github.com/MinecraftForge/MinecraftForge/pull/7645>

File: src/main/java/com/ldtteam/blockout/controls/ToggleButton.java [hunk 4]
```diff
@@ -44,8 +76,32 @@ public ToggleButton(final PaneParams params)
...
+    /**
+     * @param raw true if the result should not be localized
+     * @return the list of states as strings
+     */
+    public List<String> getStateStrings(boolean raw)
+    {
+        return raw ? rawStates : states.stream().map(IFormattableTextComponent::getString).collect(Collectors.toList());
     }
 
+    public List<IFormattableTextComponent> getStates()
```
Comment (Raycoms) [inline src/main/java/com/ldtteam/blockout/controls/ToggleButton.java:94]: jdoc

File: src/main/java/com/ldtteam/blockout/controls/AbstractTextElement.java (hunk 3 above)
Comment (Raycoms) [reply src/main/java/com/ldtteam/blockout/controls/AbstractTextElement.java:300]: kk

**Comment by Raycoms** [comment]
Looks fine for me, only missing some javadoc here and there :shipit: 

@OrionDevelopment @ShadowProtocol

File: src/main/java/com/ldtteam/blockout/controls/ToggleButton.java (hunk 4 above)
Comment (ShadowProtocol) [reply src/main/java/com/ldtteam/blockout/controls/ToggleButton.java:94]: We haven't been adding jdoc for pure getters like this one. It wouldn't really clarify anything further. Between all the getters around that's hundreds of lines of code we don't really need. There was a brief discussion on this in the original PR. This is what I said then:
> I just imagine it as if it is in an auto-genned doc page. would someone understand that this sets the text alignment based off of the method title alone? If the answer is yes then it doesn't need a jdoc IMO. You just need to be careful that you aren't doing other things in the setter that won't be described by the title alone. 
But if you're just going to chuck ``/** Sets the text alignment */`` above it then that is not worth anyone's energy scrolling past it.

But if you insist then I will.

**Pull request merged**